tests/integration/standard/test_tablets.py

```diff
@@ -0,0 +1,151 @@
+        traces = results.get_query_trace()
+        events = traces.events
+        shard_set = set()
+        for event in events:
+            LOGGER.info("%s %s %s", event.source, event.thread_name, event.description)
+            shard_set.add(event.thread_name)
+            
+        self.assertEqual(len(shard_set), 1)
+        self.assertIn('querying locally', "\n".join([event.description for event in events]))
+
+        trace_id = results.response_future.get_query_trace_ids()[0]
+        traces = self.session.execute("SELECT * FROM system_traces.events WHERE session_id = %s", (trace_id,))
+        events = [event for event in traces]
+        shard_set = set()
+        for event in events:
+            LOGGER.info("%s %s", event.thread, event.activity)
+            shard_set.add(event.thread)
+            
+        self.assertEqual(len(shard_set), 1)
+        self.assertIn('querying locally', "\n".join([event.activity for event in events]))
+    
+    def create_ks_and_cf(self):
+        self.session.execute(
+            """
+            DROP KEYSPACE IF EXISTS test1
+            """
+        )
+        self.session.execute(
+            """
+            CREATE KEYSPACE test1
+            WITH replication = {
+                'class': 'NetworkTopologyStrategy', 
+                'replication_factor': 1, 
+                'initial_tablets': 8
+            }
+            """)
+
+        self.session.execute(
+            """
+            CREATE TABLE test1.table1 (pk int, ck int, v int, PRIMARY KEY (pk, ck));
+            """)
+        
+    @staticmethod
+    def create_data(session):
+        prepared = session.prepare(
+            """
+            INSERT INTO test1.table1 (pk, ck, v) VALUES (?, ?, ?)
+            """)
+        
+        for i in range(50):
+            bound = prepared.bind((i, i%5, i%2))
+            session.execute(bound)
+
+    def query_data_shard(self, session, verify_in_tracing=True):
+        prepared = session.prepare(
+            """
+            SELECT pk, ck, v FROM test1.table1 WHERE pk = ?
+            """)
+
+        bound = prepared.bind([(2)])
+        results = session.execute(bound, trace=True)
+        self.assertEqual(results, [(2, 2, 0)])
+        if verify_in_tracing:
+            self.verify_same_shard_in_tracing(results)
+
+    def query_data_host(self, session, verify_in_tracing=True):
+        prepared = session.prepare(
+            """
+            SELECT pk, ck, v FROM test1.table1 WHERE pk = ?
+            """)
+
+        bound = prepared.bind([(2)])
+        results = session.execute(bound, trace=True)
+        self.assertEqual(results, [(2, 2, 0)])
+        if verify_in_tracing:
+            self.verify_same_host_in_tracing(results)
+
+    def test_tablets(self):
+        self.create_ks_and_cf()
+        self.create_data(self.session)
+        self.query_data_host(self.session)
+
+    def test_tablets_shard_awareness(self):
+        self.query_data_shard(self.session)
```

Comment:
I had to do this this way in order for test to pass because there is an issue with dropping keyspace when using tablets (PR to fix that: https://github.com/scylladb/scylladb/pull/14954)

Reply:
Done